gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/api/ConfigKeyPath.java

```diff
@@ -12,28 +12,45 @@
 
 package gobblin.config.store.api;
 
+import gobblin.annotation.Alpha;
+
 /**
- * The ConfigKeyPath is used to describe the relative path for a given configuration key URI to 
- * configuration store
+ * The ConfigKeyPath is used to describe the relative path for a given configuration key URI to
+ * the root URI of a config store ({@link ConfigStore#getStoreURI()}). For example,
+ * for a configuration store with root URI hfs://namenode.grid.company.com:9000/configs/hfs_config_root/
+ * and a config key URI hfs://namenode.grid.company.com:9000/configs/hfs_config_root/data/tracking/,
+ * the ConfigKeyPath will be /data/tracking.
+
  * @author mitu
  *
  */
+@Alpha
 public interface ConfigKeyPath {
 
+  /**
+   * The path to the parent.
+   * @throws IllegalArgumentException if the current node is the root.
```

Comment:
Hadoop would just throw null for this. I'm ok if we want to avoid NPE, but I don't think it should be an `IllegalArgumentException` (no arguments). What about `UnsupportedOperationException`?